File: rust/55.jump-game.rs
/*
 * @lc app=leetcode id=55 lang=rust
 *
 * [55] Jump Game
 */

// @lc code=start
impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        if nums.len() == 1 {
            return true;
        }
        let mut farther = 0;
        for i in 0..nums.len() - 1 {
            farther = farther.max(i + nums[i] as usize);
            if i == farther {
                return false;
            }
        }
        true
    }
}
// @lc code=end

File: javascript/55.jump-game.js
/*
 * @lc app=leetcode id=55 lang=javascript
 *
 * [55] Jump Game
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function (nums) {
  if (nums.length === 1) return true;
  let farther = 0;
  for (let i = 0; i < nums.length - 1; i++) {
    farther = Math.max(farther, i + nums[i]);
    if (i === farther) return false;
  }
  return true;
};
// @lc code=end
